### PR TITLE
fix: Prevent Writing Same Value to Async Storage

### PR DESCRIPTION
**Pull request by sternetj**

### CODE REVIEW

File: src/hooks/useAsyncStorage.ts [hunk 1]
```diff
@@ -6,10 +7,16 @@ export function useAsyncStorage(key: string) {
...
-    );
+  const setItem = useCallback(
+    (value: string) => {
+      if (value !== itemResult.data) {
+        AsyncStorage.setItem(key, value).catch((error) =>
+          console.error(`[LifeOmicSDK]:${error}`),
+        );
+      }
+    },
+    [key, itemResult.data],
```
Comment (markdlv) [inline src/hooks/useAsyncStorage.ts:18]: Does `itemResult.data` somehow update after one calls `setItem('value')`?  I wouldn't think so, so couldn't something like this happen?

- `itemResult.data` is `1`
- `setItem('2')` -> AsyncStorage is used
- `setItem('2')` -> AsyncStorage is used again
- `setItem('2')` -> AsyncStorage is used again

Comment (sternetj) [reply src/hooks/useAsyncStorage.ts:18]: Ahh yeah I'll test this but I bet you are correct.

File: src/hooks/useAsyncStorage.test.tsx [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+const renderHookInContext = async (key: string) => {
+  return renderHook(() => useAsyncStorage(key), {
+    wrapper: ({ children }) => (
+      <QueryClientProvider client={queryClient}>{children}</QueryClientProvider>
+    ),
+  });
+};
+
+describe('useAsyncStorage', () => {
+  test('configured appTiles are returned', async () => {
```
Comment (markdlv) [inline src/hooks/useAsyncStorage.test.tsx:23]: Copy pasta title?

**Review by markdlv — DISMISSED** [review]

File: src/hooks/useAsyncStorage.test.tsx (hunk 2 above)
Comment (sternetj) [reply src/hooks/useAsyncStorage.test.tsx:23]: 🤦‍♂️ yes

**Review by markdlv — APPROVED** [review]

**Pull request merged**